src/Blue/Tools/Api/Client.php

```diff
@@ -156,20 +156,29 @@ private function resolve(ResponseInterface $response)
 
             $attempts = $this->deferredResultMaxAttempts;
 
+            $locationHeader = $response->getHeader('location');
+
             while ($attempts > 0) {
+
+                if($locationHeader) {
```

Comment:
The location header returned by get_deferred_results would point back to get_deferred_results if the result was not ready yet.  I like the idea of making this change, but we'd also need to update all the bsd api methods that get deferred to also return a location header pointing to get_deferred_results.